### PR TITLE
Add launch your store homescreen status tour

### LINKED ISSUE #46070: [Launch Your Store] Tour
## Issue Description

Add a tour for existing users to learn about the homescreen status badge.

Image

[Figma](https://www.figma.com/file/0BWMerqRR1f468EwC7g5tg/LYS?type=design&node-id=858-51429&mode=design&t=4PLoT5Yi0FirrY1j-4)

## Acceptance Criteria

* The tour should be activated when the WooCommerce plugin is updated and the previous version was `< 8.9`. Perhaps an option needs to be set in the [db update functions](https://github.com/woocommerce/woocommerce/blob/56ca4e3f3ccb0d2c0fbaf815ba1ec1d333855fc7/plugins/woocommerce/includes/wc-update-functions.php).
* The tour should appear on the homescreen until it has been dismissed with the close button
* The tour should not appear when a site switches from coming soon to live.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #46070  .

This PR adds a tour for the homescreen site status badge.

Screen Shot 2024-04-03 at 3 39 34 PM

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with `launch-your-store` feature flag enabled.
2. Go to `WooCommerce -> Home`
3. Confirm the tour doesn't show up.

Updating from older versions of WooCommerce

1. Create a new JN site without WooCommerce.
2. Download the latest 8.6.x version [here](https://downloads.wordpress.org/plugin/woocommerce.8.6.1.zip)
3. Upload the zip file and activate it.
4. Checkout this branch locally and build a new zip file in `plugins/woocommerce` directory with `pnpm run build:zip` command
5. Upload the zip file and replace the current version with it.
6. A database update notice should show up. Click on update.
7. SSH into the server and run `wp option get woocommerce_show_lys_tour`
8. Confirm the option exists.
9. Go to `WooCommerce -> Home`
10. Confirm the tour shows up.
11. Close the tour
12. Refresh the page
13. Confirm the tour is not shown

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Add a tour for the homescreen site status badge.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/launch-your-store/tour/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+						},
+						meta: {
+							name: 'set-your-store-visibility',
+							heading: __(
+								"Set your store's visibility",
+								'woocommerce'
+							),
+							descriptions: {
+								desktop: __(
+									'Launch your store only when you’re ready to by switching between “Coming soon” and “Live” modes. Build excitement by creating a custom page visitors will see before you’re ready to go live. ',
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/launch-your-store/tour/index.tsx]: Extra space
```suggestion
									'Launch your store only when you’re ready to by switching between “Coming soon” and “Live” modes. Build excitement by creating a custom page visitors will see before you’re ready to go live.',
```

File: plugins/woocommerce/includes/class-wc-install.php [hunk 2]
```diff
@@ -247,6 +247,7 @@ class WC_Install {
 		),
 		'8.7.0' => array(
 			'wc_update_870_prevent_listing_of_transient_files_directory',
+			'wc_update_870_add_launch_your_store_tour_option',
```
Comment (adrianduffell) [inline plugins/woocommerce/includes/class-wc-install.php]: Sorry, I had the wrong version number in the issue. It should be 8.9.

Comment (ilyasfoo) [inline plugins/woocommerce/includes/class-wc-install.php]: I think this needs to be in `8.9.0` instead

File: plugins/woocommerce/includes/wc-update-functions.php [hunk 3]
```diff
@@ -2667,3 +2667,10 @@ function wc_update_870_prevent_listing_of_transient_files_directory() {
 	$wp_filesystem->put_contents( $default_transient_files_dir . '/.htaccess', 'deny from all' );
 	$wp_filesystem->put_contents( $default_transient_files_dir . '/index.html', '' );
 }
+
+/**
+ * Add woocommerce_launch_your_store_tour_hidden.
+ */
+function wc_update_870_add_launch_your_store_tour_option() {
```
Comment (ilyasfoo) [inline plugins/woocommerce/includes/wc-update-functions.php]: Same comment as above

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Would you like to add the `Discover more` link? Adrian mentioned https://github.com/woocommerce/woocommerce/issues/46070#issuecomment-2036204148 to link it to `#` first since the documentation isn't ready yet.

Tests well, but I have a comment regarding db update version number

**Comment by moon0326** [comment]
> Would you like to add the `Discover more` link? Adrian mentioned https://github.com/woocommerce/woocommerce/issues/46070#issuecomment-2036204148 to link it to `#` first since the documentation isn't ready yet.

Sounds good! Updated in [Add discover more link](https://github.com/woocommerce/woocommerce/pull/46203/commits/dfa71960ddbf343328282df327be8a8e6308e389)

File: plugins/woocommerce/includes/class-wc-install.php (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce/includes/class-wc-install.php]: Updated in [woocommerce_launch_your_store_tour_hidden should be added < 8.9](https://github.com/woocommerce/woocommerce/pull/46203/commits/7bbf8c1c03f1784f3ead2e0a3a333b99ffd647ce)

**Review by ilyasfoo — APPROVED** [review]
Minor thing - the padding at bottom is tad too much I feel:

image


Otherwise, testing well, thanks for the changes, LGTM! Pre-approving 🚢

**Comment by moon0326** [comment]
> Minor thing - the padding at bottom is tad too much I feel:
> 
> image
> Otherwise, testing well, thanks for the changes, LGTM! Pre-approving 🚢

Thanks for the suggestion 👍 Fixed in https://github.com/woocommerce/woocommerce/pull/46203/commits/b7961db0f0c9e0521393ae1055c28f9c3fd8a5ab


@ilyasfoo @adrianduffell 

I've also changed it to use user meta for woocommerce_launch_your_store_tour_hidden field in https://github.com/woocommerce/woocommerce/pull/46203/commits/8de8c83b07d70b5bdb99dbc8e79f029dfdf47f7f

File: plugins/woocommerce/includes/wc-update-functions.php [hunk 4]
```diff
@@ -2667,3 +2667,10 @@ function wc_update_870_prevent_listing_of_transient_files_directory() {
 	$wp_filesystem->put_contents( $default_transient_files_dir . '/.htaccess', 'deny from all' );
 	$wp_filesystem->put_contents( $default_transient_files_dir . '/index.html', '' );
 }
+
+/**
+ * Add woocommerce_launch_your_store_tour_hidden.
+ */
+function wc_update_890_add_launch_your_store_tour_option() {
+	update_option( 'woocommerce_launch_your_store_tour_hidden', 'no' );
```
Comment (adrianduffell) [inline plugins/woocommerce/includes/wc-update-functions.php]: The tour should only be shown on sites upgrading from a previous WooCommerce version. I think we can flip the logic here a bit and name the option `woocommerce_show_lys_tour`with a default value of `yes`. It also needs to be checked for in `useSiteVisibilityTour`

**Review by adrianduffell — CHANGES_REQUESTED** [review]
Nice work on this @moon0326, thank you. It's looking good to me but I think just needs a logic tweak to exclude the tour on new sites -- it should only appear on sites that have updated WooCommerce plugin.

I also still see the large padding at the bottom 🤔 

Screen Shot 2024-04-08 at 11 25 39 am

**Comment by chihsuan** [comment]
> Nice work on this @moon0326, thank you. It's looking good to me but I think just needs a logic tweak to exclude the tour on new sites -- it should only appear on sites that have updated WooCommerce plugin.
> 
> I also still see the large padding at the bottom 🤔

@adrianduffell Fixed padding issue in 7fc3e76996c6a398c24b24dd6de0d0136637cff5 and updated logic to exclude new sites from a tour in 24fb598e6fb7b69752466ce0e1cd108589aabbe9.

figma: 0BWMerqRR1f468EwC7g5tg-fi-858%3A49797

Actual:

Screenshot 2024-04-08 at 12 40 59

**Review by adrianduffell — APPROVED** [review]
Thanks for the latest tweaks, @chihsuan and for the changes overall, @moon032 ! It LGTM! 🚀

**Pull request merged**